### PR TITLE
Cm pam change

### PR DESCRIPTION
**Pull request by david-deboer**
This became a fairly comprehensive update to de-argify stuff, as well as put in the pam script, with a number of improvements on scripting stuff.  It is up to date with the current master.

I also started doing stuff for the new cm db workflow.  But then realized I should do that separately...

### CODE REVIEW

**Comment by david-deboer** [comment]
Uhh - sorry, I see tests aren't passing (I was over-eager on the PR)...

File: hera_mc/part_connect.py [hunk 1]
```diff
@@ -369,6 +509,8 @@ def format_check_update_connection_request(request):
     valueN:  corresponding new value
     """
```
Comment (bhazelton) [inline hera_mc/part_connect.py]: Docstring has wrong parameters

File: hera_mc/tests/test_geo_location.py [hunk 2]
```diff
@@ -17,18 +17,66 @@
...
+        gl.station_name = prefix+'_null'
+        gl.station_type_name = stn
+        gl.datum = 'WGS84'
+        gl.tile = '34J'
+        gl.northing = 6601181.0
+        gl.easting = 541007.0
+        gl.elevation = 1051.69
+        gl.created_gpstime = 1172530000
+        self.test_session.add(gl)
+        self.test_session.commit()
```
Comment (bhazelton) [inline hera_mc/tests/test_geo_location.py]: This cofa adding code looks like it's copied and pasted a bunch of places. You might consider making a function in `tests/__init__.py` that contains this code and could be called quickly from the appropriate tests.

File: hera_mc/tests/test_geo_location.py [hunk 3]
```diff
@@ -17,18 +17,66 @@
 
 class TestGeo(TestHERAMC):
 
-    def test_add_part(self):
-        self.test_session.add(geo_location.GeoLocation())
+
+    def test_add_loc(self):
```
Comment (bhazelton) [inline hera_mc/tests/test_geo_location.py]: I'm not sure this test is adding anything, given that your 3rd test does exactly the same thing and then retrieves the value and checks that it is as expected.

File: scripts/pam_swap.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
+#! /usr/bin/env python
+# -*- mode: python; coding: utf-8 -*-
+# Copyright 2016 the HERA Collaboration
+# Licensed under the 2-clause BSD license.
+
+"""
+Script to handle swapping of PAMs.
+"""
```
Comment (bhazelton) [inline scripts/pam_swap.py]: I worry about how much logic is in this script and therefore very difficult to test. Could you make it a very thin (almost no if/else logic) wrapper around a function that could be tested?

File: hera_mc/mc.py [hunk 5]
```diff
@@ -110,6 +110,20 @@ def get_mc_argument_parser():
                    help='Name of the database to connect to. The default is used if unspecified.')
     return p
 
+def get_cm_csv_path(args):
```
Comment (david-deboer) [inline hera_mc/mc.py]: I'll finish this shortly.  It does not currently exist.  I plan to do this tomorrow.

File: scripts/pam_swap.py (hunk 4 above)
Comment (david-deboer) [inline scripts/pam_swap.py]: These scripts are going to need a bit of separate/specific logic.  I have pulled out some stuff that is more generic.

Comment (bhazelton) [inline scripts/pam_swap.py]: It's not just about pulling out generic code, even specific code can be in a module that can be tested. The problem is if you have a lot of logic in scripts, there's no good way to actually test the logical branches.

File: hera_mc/geo_handling.py [hunk 6]
```diff
@@ -51,10 +49,8 @@ def get_location(location_names, query_date='now', show_location=False, verbosit
     location_name:  location name, may be either a station (geo_location key) or an antenna
```
Comment (bhazelton) [inline hera_mc/geo_handling.py]: location_name -> location_names to be consistent with function definition.

File: scripts/pam_swap.py (hunk 4 above)
Comment (bhazelton) [inline scripts/pam_swap.py]: per Slack conversation, this script is fairly temporary to implement a specific change, so doesn't need the same level of testing.

File: hera_mc/geo_handling.py [hunk 7]
```diff
@@ -31,15 +31,13 @@ def cofa(show_cofa=False):
     show_cofa:  boolean to print out cofa info or just return class
```
Comment (bhazelton) [inline hera_mc/geo_handling.py]: resolved.

File: hera_mc/geo_handling.py [hunk 8]
```diff
@@ -95,9 +91,16 @@ def close(self):
     def cofa(self, show_cofa=False):
```
Comment (bhazelton) [inline hera_mc/geo_handling.py]: resolved.

File: hera_mc/cm_handling.py [hunk 9]
```diff
@@ -35,293 +36,265 @@ class Handling:
...
-                self.show_other_connections(connection_dict, already_shown)
-        if return_dictionary:
-            return connection_dict
-
-    def show_active_connections(self, connection_dict):
+        #-#nc = self.no_connection_designator
+        #-#connection_dossier[nc] = PC.Connections(upstream_part=nc, upstream_output_port=nc, up_part_rev=nc,
+        #-#                                               downstream_part=nc, downstream_input_port=nc, down_part_rev=nc,
+        #-#                                               start_gpstime=cm_utils._get_datetime('<', '<').gps,
+        #-#                                               stop_gpstime=cm_utils._get_datetime('>', '>').gps)
```
Comment (david-deboer) [inline hera_mc/cm_handling.py]: Removed.

Comment (bhazelton) [inline hera_mc/cm_handling.py]: I think they're still there

File: hera_mc/tests/test_geo_location.py (hunk 2 above)
Comment (david-deboer) [inline hera_mc/tests/test_geo_location.py]: I'm not sure.  It is nice to control from within your module.  I'd like to defer this one to think about it.

**Comment by david-deboer** [comment]
The csv stuff is handled in the next PR.

**Pull request merged**